### PR TITLE
[ADP-3212] restructure delegation certificate

### PR DESCRIPTION
**Pull request by paolino**
- [x] RemoveCertRegisterKey constructor from  DelegationCertificate 

ADP-3212

### CODE REVIEW

**Review by HeinrichApfelmus — APPROVED** [review]
Thanks! 😊

File: lib/wallet/api/http/Cardano/Wallet/Api/Types/Certificate.hs [hunk 1]
```diff
@@ -311,7 +311,7 @@ mkApiAnyCertificate acct' acctPath' = \case
         else
             DelegationCertificate
                 $ QuitPoolExternal (ApiRewardAccount rewardKey)
-    toApiDelCert acctM acctPath (W.CertRegisterKey rewardKey) =
+    toApiDelCert acctM acctPath (W.CertVoteAndDelegate rewardKey Nothing Nothing) =
```
Comment (paweljakubas) [inline lib/wallet/api/http/Cardano/Wallet/Api/Types/Certificate.hs:314]: good idea to utilize Nothing Nothing case !

**Review by paweljakubas — APPROVED** [review]
LGTM

File: lib/wallet/api/http/Cardano/Wallet/Api/Types/Certificate.hs (hunk 1 above)
Comment (paolino) [reply lib/wallet/api/http/Cardano/Wallet/Api/Types/Certificate.hs:314]: not an idea, just what the ledger say in Conway

**Pull request merged**